src/adr/0035-distributed-listeners-configuration.md

```diff
@@ -0,0 +1,145 @@
+An event sourcing system will track the configured mailbox listeners.
+
+It will have the following commands:
+
+ - **AddListener**: Add a given listener. This should be rejected if the group is already used.
+ - **RemoveListener**: Remove a given listener.
+
+It will have the following events:
+
+ - **ListenerAdded**: A mailbox listener is added
+ - **ListenerRemoved**: A mailbox listener is removed
+
+A subscriber will react to these events to modify the RabbitMQ resource accordingly by adding queues, adding or removing
+bindings.
+
+This event sourcing system differs from the one defined in
+[26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md) by the
+fact that we should also keep track of listener configuration.
+
+Upon start, James will ensure the **configured mailbox listener event sourcing system** contains the guice injected 
+listeners, and add them if missing (handling the RabbitMQ bindings by this mean), then starts the eventBus which will
+consume the given queues.
+
+If a listener is configured with a class unknown to James, the start-up fails and James starts in a degraded state 
+allowing to unconfigure the faulty listener. This downgraded state will be described in a separate ADR and the link will
+be updated here.
+
+This differs from [26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md)
+by the fact we no longer need to register all listeners at once.
+
+A WebAdmin endpoint will allow:
+
+ - **to add a listener** to the one configured. Such a call:
+    - Will fail if the listener class is not on the local classpath, or if the corresponding group is not part of the
+     configured listeners aggregate.
+    - Upon success the listener is added to the **configured mailbox listener aggregate**, and the listener is 
+    registered locally.
+ - **to remove a listener**. Such a call:
+    - Will fail if the listener is required by Guice bindings on the current server or if the listener is not configured.
+    - Upon success, the listener is removed from the **configured mailbox listener aggregate**, and the listener is 
+    unregistered locally.
+    
+A broadcast on the event bus will be attempted to propagate topology changes, by the mean of a common registrationKey 
+to all nodes, a "TopologyChanged" event, and a mailbox listener starting the MailboxListeners on local node upon
+topology changes.
+ 
+If a listener is added but is not in the classpath, an ERROR log is emitted. This can happen during a rolling upgrade,
+which defines a new guice binding for a new mailbox listener. Events will still be emitted (and consumed by other James
+servers) however a local James upgrade will be required to effectively be able to start processing these events. The 
+binding will not need to be redefined.
+
+Integration tests relying on additional mailbox listeners of the distributed James product will require to be ported to 
```

Comment:
should be consequences?

Reply:
Sorry I don't understand that one...

Reply:
I mean it may be in "consequences" section of this ADR